### ISSUE TITLE
Setting background color on block changes custom gradient attribute 

### ISSUE THREAD

**Issue by andreaslindahl** [WordPress/gutenberg#53941]
### Description

Starting in WP 6.3 I'm experiencing a bug related to block background color and attributes in my block.

I have a custom block with an attribute named "gradient" (boolean). Supports for the block are configured as follows:

```
"color": {
	"background": true,
	"text": true
}	
```

If the gradient attribute for my block is set to true, it is automatically set to false when a background color is selected in the editor. This used to work fine in previous versions of Wordpress. My guess is that my custom gradient attribute conflicts with an internal Gutenberg attribute applied to blocks also named gradient. 

My gradient attribute is not related to the selected background color in any way, so having Gutenberg automatically change it is not exactly good in my case...

First of all, is there a way to fix this without migrating all of my blocks to use another attribute name for my gradient setting?

And is there a list of reserved attribute names used by Gutenberg to avoid future conflicts with other attribute names?

### Step-by-step reproduction instructions

Create a custom block with a boolean attribute named gradient.
Add support for background color in the block.json file.
In the editor, make sure that the gradient attribute is set to true
Select a background color
The gradient attribute is set to false

### Environment info

Wordpress 6.3

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by tellthemachines** [comment]
> My guess is that my custom gradient attribute conflicts with an internal Gutenberg attribute applied to blocks also named gradient.

That's correct, there is a gradient attribute that's applied to blocks when they have enabled support for color.gradients and a gradient preset is selected. It's likely that the change in behaviour you're seeing is a result of the color panel refactor in #48893, but I'm a bit surprised it worked without issue before 😅. I'm guessing you didn't have support for gradients enabled in your custom block? 

I suspect the change that broke things was [all the color attributes being set in one go](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/color.js#L221), regardless of which color settings actually changed. 

> First of all, is there a way to fix this without migrating all of my blocks to use another attribute name for my gradient setting?

I'm not sure. It's likely we'll want to keep the current logic in order to reset any attributes that no longer apply. I've [asked](https://github.com/WordPress/gutenberg/pull/48893/files#r1550924409) for confirmation in the original PR.

> And is there a list of reserved attribute names used by Gutenberg to avoid future conflicts with other attribute names?

Unfortunately there isn't such a list, but it would be great to add one to the handbook!  The closest I can think of is looking through the list of[ block supports](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/) for names already in use there.

**Comment by andreaslindahl** [comment]
That's probably the reason. However, it would be better - at least for me - to only change attributes for settings that the block actually supports, instead of setting them all automatically.
